05/app.js

```diff
@@ -1,37 +1,69 @@
-const apiUrl = 'http://localhost:3000/users';
+const apiUrl = 'http://localhost:3000/users'
 
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener('DOMContentLoaded', init)
 
 function init() {
-    loadUsers();
+	const formEl = document.querySelector('form')
+	if (formEl) {
+		formEl.addEventListener('submit', handleSubmit)
+	}
+	loadUsers()
 }
 
 function loadUsers() {
-    const promise = fetchGet(apiUrl);
+	const promise = fetchGet(apiUrl)
 
-    promise
-        .then(data => insertUsers(data))
-        .catch(err => console.error(err));
+	promise.then((data) => insertUsers(data)).catch((err) => console.error(err))
 }
 
 function fetchGet(url) {
-    return fetch(url)
-        .then(resp => {
-            if(resp.ok) {
-                return resp.json();
-            }
-
-            return Promise.reject(resp);
-        });
+	return fetch(url).then((resp) => {
+		if (resp.ok) {
+			return resp.json()
+		}
+
+		return Promise.reject(resp)
+	})
 }
 
 function insertUsers(usersList) {
-    const ulElement = document.querySelector('.users');
-    ulElement.innerHTML = '';
-    usersList.forEach(user => {
-        const liElement = document.createElement('li');
-        liElement.innerText = `${user.firstName} ${user.lastName}`;
-
-        ulElement.appendChild(liElement);
-    });
+	const ulElement = document.querySelector('.users')
+	ulElement.innerHTML = ''
+	usersList.forEach((user) => {
+		const liElement = document.createElement('li')
+		liElement.innerText = `${user.firstName} ${user.lastName}`
+
+		ulElement.appendChild(liElement)
+	})
+}
+
+function handleSubmit(e) {
+	e.preventDefault()
+
+	const firstName = e.target.elements[0]
+	const lastName = e.target.elements[1]
```

Comment:
Jeśli mamy gdzieś [0] itp. to robimy destrukturyzacje - jest czytelniej :) czyli:
```
const [firstName, lastName] = e.target.elements;
```

Reply:
Fakt 👍  poprawione